File: src/test/resources/sample_module/io/github/grimch/doclet/sample_module/types/utility/C5.java
/*
 * This file is part of LogiDoclet.
 *
 * Copyright (c) 2025 The LogiDoclet Authors
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package io.github.grimch.doclet.sample_module.types.utility;

/**
 * Represents a package-private utility class that is {@link Deprecated}.
 * This class demonstrates a static package-private field and a synchronized method.
 * It is intended for internal use within its package.
 * @deprecated This class is deprecated and should not be used.
 */
@Deprecated
class C5 {
    /**
     * A static package-private integer field.
     */
    static int PACKAGE_PRIVATE_FIELD = 5;

    /**
     * A public synchronized method.
     * This method ensures thread-safe execution.
     */
    public synchronized void m15() {
        // Implementation
    }
}

File: src/test/resources/expected_output/full/io/github/grimch/doclet/sample_module/types/utility/C5.pl
class(
    'C5',
    'io.github.grimch.doclet.sample_module.types.utility',
    [],
    [],
    'null',
    [],
    [],
    [
        constructor('<init>', [], [], [], [], [], ''),
        field(
            'PACKAGE_PRIVATE_FIELD',
            [
                modifier(static)
            ],
            type(primitive, int),
            [],
            ' A static package-private integer field.\n'
        ),
        method(
            m15,
            [
                modifier(public),
                modifier(synchronized)
            ],
            [],
            type(no_type, void),
            [],
            [],
            [],
            ' A public synchronized method.\n This method ensures thread-safe execution.\n'
        )
    ],
    [
        annotation('java.lang.Deprecated', [])
    ],
    ' Represents a package-private utility class that is {@link Deprecated}.\n This class demonstrates a static package-private field and a synchronized method.\n It is intended for internal use within its package.\n @deprecated This class is deprecated and should not be used.\n'
).
